### PR TITLE
complete handle each command error

### PR DESCRIPTION
**Pull request by jjaykim**
# Befor putting PR (put x in bracket if you ran it.);

- [x] Run `pylint $(git ls-files '*.py')` passes

# Overview

Issue  https://github.com/dhmoon91/discord/issues/9
For each commend, handle possible errors and if any send a message.

A situation an error can occur in `get_rank`
- if the user inputs an invalid summoner name
- get_rank command only accepts one summoner name
- general API error eg. Rate limit exceeded or Service unavailable etc.

A situation an error can occur in `get_last_match`:
- if the user inputs an invalid summoner name
- get_last_match command only accepts one summoner name
- general API error eg. Rate limit exceeded or Service unavailable etc.

**UPDATE on July 11**
consider general error cases in the help command

The message of when summoner name not found or inputs more then one summoner name
- desktop version 
Screen Shot 2021-07-11 at 12 35 14 PM

- mobile version
Screen Shot 2021-07-11 at 12 43 21 PM

The message of when occurs other cases of an error
(The general error message is the same case as @jung1411's add_summoner, so I brought a sentence)
- desktop version
Screen Shot 2021-07-11 at 12 35 20 PM

- mobile version
Screen Shot 2021-07-11 at 12 37 48 PM

### CODE REVIEW

File: bot.py
Comment (dhmoon91) [inline bot.py]: why is this here?

Comment (dhmoon91) [reply bot.py]: oh I see why. you are using it to catch

File: bot.py
Comment (dhmoon91) [inline bot.py]: hmm I don't like this because you are just looking at `ApiError`, but I see why. Did you think of any other possible cases that we should think about?

File: bot.py
Comment (jjaykim) [reply bot.py]: Yes. Right!

File: bot.py
Comment (dhmoon91) [reply bot.py]: do you still need `as e_values`

File: bot.py
Comment (dhmoon91) [inline bot.py]: I think we can put in `Please type `rank --help` to see how to use` here as well

Comment (jjaykim) [reply bot.py]: Okay. I will add it

File: bot.py
Comment (jjaykim) [reply bot.py]: The reason why I used `ApiError` is because
1. I ignore the space when the user inputs a summoner name and receive it as one summoner name
eg. `!rank life is so hard d4022` (the user try to search two summoner name at a time) but I receive as `life is so hard d4022` and then send it API to find. Finally, API will send `ERROR` due to the `life is so hard d4022` summoner name is not found. 
-> if more than one user is searched or an invalid user name is entered, check it at once.

2. Although the user entered a valid summoner name, `ApiError` was used to prepare for other API errors.

I need `e_values` cuz to check whether the summoner name is not found or other API errors like `if e_values.response.status_code == 404:`

Comment (jjaykim) [reply bot.py]: Finally, I've understood using` Exception` is better than `ApiError`. I tried to use `Exception` to catch any other possible errors.

File: bot.py
Comment (HelloFive) [inline bot.py]: What does this look like on discord? `error_description` seems pretty long..
This can be shortened into 
```
error_description = "Oops! Something went wrong.\n\n Please type `last_match --help` to see how to use and try again!"
```

File: bot.py
Comment (HelloFive) [inline bot.py]: description can be shorted just like https://github.com/dhmoon91/discord/pull/27/files#r668373752

File: bot.py
Comment (HelloFive) [inline bot.py]: Same here. `error_description` can be shorted like https://github.com/dhmoon91/discord/pull/27/files#r668373752

File: bot.py
Comment (jjaykim) [reply bot.py]: The image shown in discord is attached above.

The reason why I use `Please try again` is that I considered situations where the user input correctly but there could be a system error. However, if you think `Please try agian` doen not need, I will take it out.

Also, `\n\n` which follows `\`(line #199) needs to fix the mobile version of indentation. 
if I use only one `\n` the embed will be shown like
Screen Shot 2021-07-13 at 3 44 13 AM

Lastly, `\n\n` which follows `something went wrong` is my mistake. Only one `\n` needs in there.

This explanation will be also the answer to the two reviews below.
https://github.com/dhmoon91/discord/pull/27#discussion_r668374187
https://github.com/dhmoon91/discord/pull/27#discussion_r668374501

**Review by dhmoon91 — APPROVED** [review]
no blocker from me

**Review by jung1411 — APPROVED** [review]
Looks good to me

**Comment by dhmoon91** [comment]
@jjaykim merge this if you think it's good to go you have two approvals

**Comment by jjaykim** [comment]
> @jjaykim merge this if you think it's good to go you have two approvals

Yes. I think it's good and also we discussed with this pr on last Tuesday.
Should I click squash and merge or will you do it?

**Comment by dhmoon91** [comment]
@jjaykim this is your pr dont wait for other people to merge it for you lol

**Pull request merged**
